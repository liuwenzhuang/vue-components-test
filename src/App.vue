<template>
  <div id="app">
    <MessageList :messages="messages" />
    <Form />
  </div>
</template>

<script>
import MessageList from './components/MessageList.vue'
import Form from './components/Form.vue'

export default {
  name: 'App',
  components: {
    MessageList,
    Form
  },
  data() {
    return {
      messages: ['Msg1', 'Msg2']
    }
  }
}
</script>

<style lang="less">
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
