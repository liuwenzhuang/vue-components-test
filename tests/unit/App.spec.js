import { shallowMount } from '@vue/test-utils'
import App from '@/App.vue'

describe('App.vue', () => {
  let cmp
  const messages = ['New Msg1', 'New Msg2']
  beforeEach(() => {
    cmp = shallowMount(App, {
      data() {
        return { messages }
      }
    })
  })
  it('messages should equal messages', () => {
    expect(cmp.vm.messages).toEqual(messages)
  })
  it('should match html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })
})
