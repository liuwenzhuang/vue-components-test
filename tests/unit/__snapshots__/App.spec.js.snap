// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should match html structure 1`] = `
<div
  id="app"
>
  <messagelist-stub
    messages="New Msg1,New Msg2"
  />
   
  <form-stub />
</div>
`;
